### ISSUE TITLE
An edge collapses into a point when the edge is too short 

### ISSUE THREAD

**Issue by dareumnam** [NREL/EnergyPlus#7442]
Issue overview
--------------

Related to #7372 
Link to the idf file (provided by Mitchal): 
https://drive.google.com/file/d/1LUbjQIDpn5CM5l-mkz2rSFO4jv-IGleP/view?usp=sharing

It seems like E+ makes an edge of a building model geometry disappear when the edge is too short (according to the last warning message below). The geometry in the IDF file is not fully enclosed and has a very small ceiling height (0.01m). When simulated, the error file says:
```
   ** Warning ** CalculateZoneVolume: The Zone="ROOM 102" is not fully enclosed. To be fully enclosed, each edge of a surface must also be an edge on one other surface.
   **   ~~~   **   The zone volume was calculated using the floor area times ceiling height method where the floor and ceiling are the same except for the z-coordinates.
   **   ~~~   **   The surface    "ROOM102 SOUTH WALL" has an edge that is either not an edge on another surface or is an edge on three or more surfaces: 
   **   ~~~   **     Vertex start { 0.0000, 0.0000, 0.0000}
   **   ~~~   **     Vertex end   { 3.6739E-014, -300.0000, 1.0000E-002}
   **   ~~~   **   The surface    "ROOM102 SOUTH WALL" has an edge that is either not an edge on another surface or is an edge on three or more surfaces: 
   **   ~~~   **     Vertex start { 0.0000, 0.0000, 0.0000}
   **   ~~~   **     Vertex end   { 0.0000, 0.0000, 0.0000}
TRUNCATED
```
If this is the case, it seems like an error or warning message should be generated when an edge collapses into a point. 

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.0

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (https://drive.google.com/file/d/1LUbjQIDpn5CM5l-mkz2rSFO4jv-IGleP/view?usp=sharing)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by JasonGlazer** [comment]
@dareumnam this is as designed.  A tolerance is included when testing if points are equal and currently that tolerance is 1/2 inch (1.27 cm) which appears in:

https://github.com/NREL/EnergyPlus/blob/53b18e4aa9f4c3b539df6fd4d012437b360e6570/src/EnergyPlus/SurfaceGeometry.cc#L10542

So your example file fails. I guess I don't understand the case you are trying to simulate that has a wall height of just 1 cm. Could you explain further?

What tolerance would be reasonable?

**Comment by dareumnam** [comment]
@JasonGlazer You're right. That makes sense since the example idf file has 1cm of wall heights. The 1/2 inch (1.27 cm) of tolerance sounds reasonable enough to me. I think if the author of the original issue (#7372) confirms this, this issue can be closed. @mldichter

**Comment by mldichter** [comment]
@dareumnam @JasonGlazer The walls are 1cm tall in the model, but the strange part is EnergyPlus appears to collapse two points then treat the two collapsed points as an edge still.

Vertex 3 and 4 are very close together, and the error suggests that the Vertex 4 Z-coordinate {m} was changed from 0.01 to 0.0000, which would collapse Vertex 4 to Vertex 3. The error then references an edge from the origin to the origin.

```
The surface    "ROOM102 SOUTH WALL" has an edge that is either not an edge on another surface or is an edge on three or more surfaces: 
Vertex start { 0.0000, 0.0000, 0.0000}
Vertex end   { 0.0000, 0.0000, 0.0000}

BuildingSurface:Detailed,
    Room102 South Wall,      !- Name
    WALL,                    !- Surface Type
    Room Wall - South,       !- Construction Name
    Room 102,                !- Zone Name
    Outdoors,                !- Outside Boundary Condition
    ,                        !- Outside Boundary Condition Object
    SunExposed,              !- Sun Exposure
    WindExposed,             !- Wind Exposure
    0.5,                     !- View Factor to Ground
    4,                       !- Number of Vertices
    0,                       !- Vertex 1 X-coordinate {m}
    300,                     !- Vertex 1 Y-coordinate {m}
    0.01,                    !- Vertex 1 Z-coordinate {m}
    0,                       !- Vertex 2 X-coordinate {m}
    300,                     !- Vertex 2 Y-coordinate {m}
    0,                       !- Vertex 2 Z-coordinate {m}
    0,                       !- Vertex 3 X-coordinate {m}
    0,                       !- Vertex 3 Y-coordinate {m}
    0,                       !- Vertex 3 Z-coordinate {m}
    0,                       !- Vertex 4 X-coordinate {m}
    0,                       !- Vertex 4 Y-coordinate {m}
    0.01;                    !- Vertex 4 Z-coordinate {m}
```

**Comment by JasonGlazer** [comment]
@mldichter the warning can be ignored because it is under this 1/2 inch limit and the routine that generates that warning doesn't realize it is two different points anymore. As far as the rest of EnergyPlus, I don't think anywhere else uses that 1/2 inch limit so the rest of the calculations would  be fine. Nothing is really collapsing in the model, the reporting of the vertex start and end is just wrong when they are smaller than this tolerance. I'm reluctant to fix the issue of how it is reported since it is a very unusual case. Would you like to explain further why you modeled it?

**Comment by mldichter** [comment]
@JasonGlazer It was a strange simulation. We were trying to validate an EnergyPlus model of a heatpump from field data. In order to prescribe the return air temperature I wanted to test the model against, I kept the volume of the room constant but gave it a huge amount of surface area by flattening the room and setting a boundary temperature on the surfaces. I got this weird error, tried it out in a EnergyPlus example file to make sure I could reproduce the error in a error free file, then posted an issue.

I was actually hoping that the error might have something to do with the code that determines whether or not edges of different surfaces are the same. EnergyPlus has difficulty determining whether edges touch when one edge is very long and another edge is short. For example, if the asterisks are vertices, then EnergyPlus will say that the middle edge on the bottom three does not match up with the long top edge. This has happened to me before in hotels with lots of room edges bordering a long corridor. The only way I can get the errors to go away is to split up the corridor floor surface so that none of the edges of the corridor are very long.

```
*---------------------------------------------------------------------------*
*-------------------------------*------------*------------------------------*
```
